addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -383,21 +424,48 @@ const dragAndDropCommonMenuEditor = Widget.extend({
         // grid and the dropzone.
         this.dragState.dragHelperEl.remove();
         this.dragState.backgroundGridEl.remove();
-        this.options.wysiwyg.odooEditor.observerActive("dragAndDropMoveSnippet");
+        this.options.wysiwyg.odooEditor.observerActive(this.observerName);
         gridUtils._resizeGrid(rowEl);
-        this.options.wysiwyg.odooEditor.observerUnactive("dragAndDropMoveSnippet");
+        this.options.wysiwyg.odooEditor.observerUnactive(this.observerName);
         this.handleDragMove = false;
     },
+    /**
+     * Stores the width and height characteristics of a grid item element.
+     *
+     * @private
+     * @param {Element} dropzoneEl
+     */
+    _storeGridItemWidthAndHeight(dropzoneEl) {
+        // This is hacky: The idea is that the snippet is dropped on the grid,
+        // however, as it is on the grid, it will try to fit in one box. This is
+        // a problem to define the dimensions of the drag helper. The hack is to
+        // insert the snippet on the DOM but not on the grid in order to extract
+        // its correct dimensions.
+        this.draggedItemEl.remove();
+        dropzoneEl.parentNode.after(this.draggedItemEl);
+        this._storeItemWidthAndHeight(gridUtils._getGridProperties(dropzoneEl.parentNode));
+        this.draggedItemEl.remove();
+        dropzoneEl.after(this.draggedItemEl);
+    },
     /**
      * Stores the dragged item width and height.
      * @private
+     * @param {Object} gridProp The grid properties. If set, the item comes from
+     * the right panel.
      */
-    _storeItemWidthAndHeight() {
+    _storeItemWidthAndHeight(gridProp = false) {
         const isImageColumn = gridUtils._checkIfImageColumn(this.draggedItemEl);
         // Store the image width and height if the column only contains an
         // image.
         const measuredEl = isImageColumn ? this.draggedItemEl.querySelector("img") : this.draggedItemEl;
-        this.dragState.columnWidth = parseFloat(measuredEl.scrollWidth);
+        if (gridProp) {
+            // The item comes from the right panel. Do not set it too wide on
+            // the grid.
+            this.dragState.columnWidth = Math.min(parseFloat(measuredEl.scrollWidth),
+                6 * (gridProp.columnSize + gridProp.columnGap) - gridProp.columnGap);
```

Comment:
As we do not want an inner snippet coming from the right panel to have a width bigger than 6 columns